### DISCUSSION TITLE
Calculate width of rotated image (cross post from Magick.NET!)

### DISCUSSION THREAD

**Discussion by Lemmes** [ImageMagick/ImageMagick#2097] [Help] [answered]
Using **Magick.NET-Q8-OpenMP-x64 v7.17.0.0**

Hello, 
I need to recalculate values of worldfiles for arbitrarily rotated images without having to open the rotated image and read out the actual width.
Say we have a square image with 10000 px width and height that's being rotated clockwise 1.422346555 rad. Calculation of the new dimension should be easy enough:
``` csharp
var newLength = Math.Sin(1.422346555) * 10000 + Math.Cos(1.422346555) * 10000;
Console.WriteLine(newLength);
// returns 11369,066856491
```
Of course there is no hundredth of a pixel so it has to be rounded up. But actually the rotated image is **11372** instead of **11370** pixels wide. Another image **5000** px wide resulted in a **5686** px wide rotated image with the same angle - which is **one** extra pixel after rounding up. 

So the question is: how does ImageMagick calculate the dimensions of a rotated image?
Thanks for your patience, Bernd

**Comment by snibgo** [comment, marked as answer]
Short answer: there is more than one rounding-up.

Long answer: see distort.c function DistortImage().

**Reply by Lemmes** [reply]
Thanks, took a short look at it, but I'm afraid it's beyond my C-capabilities. So I'll probably have no choice but to open the image and look at the dimensions.
